kotlin-reference-server/src/main/kotlin/org/stellar/reference/Mutex.kt

```diff
@@ -0,0 +1,23 @@
+package org.stellar.reference
+
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.retryWhen
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+// This is to make sure transaction submission is mutual exclusive to avoid failures
+internal val submissionLock = Mutex()
+
+suspend fun transactionWithRetry(transactionLogic: suspend () -> Unit) =
```

Comment:
Isn't this a duplicate?

Reply:
Yeah, it's implemented twice because it needs to be used in the reference server as well as the tests.

Reply:
Can we add to the `lib-util` sub-project?

Reply:
Can we delay this until we have more shared code? I'm not sure if we want to create a whole new project just to hold a function.